internal/component/pyroscope/scrape/manager.go

```diff
@@ -27,17 +30,18 @@ type Manager struct {
 	triggerReload chan struct{}
 }
 
-func NewManager(appendable pyroscope.Appendable, logger log.Logger) *Manager {
+func NewManager(o []config_util.HTTPClientOption, appendable pyroscope.Appendable, logger log.Logger) *Manager {
```

Comment:
I would prefer we define a struct `Options` a bit like [`prometheus/scrape`](https://github.com/prometheus/prometheus/blob/d15869af32f918b9a958c3a73144f4834c935a75/scrape/manager.go#L85-L86) does.

For now it is good enough for that struct to only have that single field of `HTTPClientOptions`

Reply:
Done in HEAD commit :D